src/NHSD.BuyingCatalogue.Identity.Api/Services/ProfileService.cs

```diff
@@ -61,26 +62,29 @@ public async Task GetProfileDataAsync(ProfileDataRequestContext context)
             if (user is not null)
             {
                 var claims = GetClaimsFromUser(user);
-                var primaryOrganisationName = await GetPrimaryOrganisationName(user.PrimaryOrganisationId);
-                if (primaryOrganisationName is not null)
-                {
-                    claims.Add(primaryOrganisationName);
-                }
+                claims.AddRange(await GetPrimaryOrganisationClaims(user.PrimaryOrganisationId));
 
                 context.IssuedClaims = claims;
             }
         }
 
-        public async Task<Claim> GetPrimaryOrganisationName(Guid primaryOrganisationId)
+        public async Task<IEnumerable<Claim>> GetPrimaryOrganisationClaims(Guid primaryOrganisationId)
         {
-            Claim claim = null;
-            var organisationName = (await organisationRepository.GetByIdAsync(primaryOrganisationId))?.Name;
-            if (!organisationName.IsNullOrEmpty())
+            List<Claim> claims = new List<Claim>();
+
+            var primaryOrganisation = await organisationRepository.GetByIdAsync(primaryOrganisationId);
+
+            if (primaryOrganisation is not null)
             {
-                claim = new Claim(ApplicationClaimTypes.PrimaryOrganisationName, organisationName!);
+                if (!primaryOrganisation.Name.IsNullOrEmpty())
+                {
+                    claims.Add(new Claim(ApplicationClaimTypes.PrimaryOrganisationName, primaryOrganisation.Name));
+                }
+
+                claims.AddRange(primaryOrganisation.RelatedOrganisations.Select(ro => new Claim(ApplicationClaimTypes.RelatedOrganisationId, ro.OrganisationId.ToString())).ToArray());
```

Comment:
Do we need the `ToArray` here?

The line is starting to get a little long too so might be worth moving the lambda a new line.

Reply:
yeah i can remove the too array and list the lambdas vertically

Reply:
tried to make this line a little neater for you, and frankly can't find one i like. best i got is this:
```c#
                claims
                    .AddRange(
                    primaryOrganisation.RelatedOrganisations.Select(
                    ro => new Claim(
                                ApplicationClaimTypes.RelatedOrganisationId,
                                ro.OrganisationId.ToString())));
```
but i'm not convinced :/

Reply:
actually got something i can live with, just gona resolve this